beacon-chain/core/light-client/lightclient.go

```diff
@@ -299,46 +329,36 @@ func ComputeWithdrawalsRoot(payload interfaces.ExecutionData) ([]byte, error) {
 	return withdrawalsRoot, nil
 }
 
-func BlockToLightClientHeader(block interfaces.ReadOnlySignedBeaconBlock) (*ethpbv2.LightClientHeaderContainer, error) {
-	switch block.Version() {
+func BlockToLightClientHeader(ver int, block interfaces.ReadOnlySignedBeaconBlock) (interfaces.LightClientHeader, error) {
```

Comment:
As discussed, I added a version parameter which is populated from the attested block's version in `NewLightClientUpdateFromBeaconState`

Reply:
Could this potentially cause a problem? for example when version is Capella but block is Altair. will calling the `block.execution` getters on an Altair block result in an error?